luigi/worker.py

```diff
@@ -511,11 +513,10 @@ def __init__(self, scheduler=None, worker_id=None, worker_processes=1, assistant
         self.worker_processes = int(worker_processes)
         self._worker_info = self._generate_worker_info()
 
-        if not worker_id:
-            worker_id = 'Worker(%s)' % ', '.join(['%s=%s' % (k, v) for k, v in self._worker_info])
-
         self._config = worker(**kwargs)
 
+        worker_id = worker_id or self._config.id or 'Worker(%s)' % ', '.join(['%s=%s' % (k, v) for k, v in self._worker_info])
```

Comment:
since you've already modified this line, any objection to going ahead and replacing that ugly `%` string interpolation with `.format()`? :)

Reply:
I would also not do a one liner for this, so it's more readable